core/procs.go

```diff
@@ -1103,11 +1103,27 @@ var procReadString Proc = func(args []Object) Object {
 	return readFromReader(strings.NewReader(EnsureString(args, 0).S))
 }
 
+func readLine(r *BufferedReader) (s string, err error) {
+	for {
+		line, isPrefix, err := r.ReadLine();
```

Comment:
Would not using `ReadString('\n')` instead result in a simper implementation?

Reply:
Looks like it, thanks! I'll play around with that more when I have some free time, maybe later today. It'd be great to avoid adding my `readLine()` function, if it isn't needed.

Reply:
Upon further review, it seems `ReadString('\n')` is not equivalent, because `ReadLine()` accounts for a potential `\r` preceding the `\n`.

Would you prefer the current implementation (in this PR), or one that replaces my implementation of `readLine()` with one that calls `ReadString('\n')` and then removes any final `\n` and, if so removed, any resulting final `\r`, before returning the resulting string?

I'm not 100% sure the results would be exactly the same in all cases (the `bufio.ReadLine()` code does some non-trivial stuff), and in general I'd prefer to leave it to Go's runtime to decide what constitutes a "line" of text (while keeping the underlying buffer "in sync" so as to handle subsequent calls to `ReadLine()` or anything else that reads from it)...but I'm open to eliminating the current loop in favor of having Go's runtime loop for us and then just cleaning up afterwards.

(I guess we can be grateful we don't have to support systems using only `\r` to end lines -- or, for that matter, EBCDIC....)

Reply:
I think current implementation is fine if the behavior is consistent with Clojure. Merging.